### DISCUSSION TITLE
Speed?

### DISCUSSION THREAD

**Discussion by TornaxO7** [wbthomason/packer.nvim#261] [Q&A]
Would you say that packer.nvim has a faster loading time than the plugin managers like vim-plug? Because currently my plugins are influencing the loading time quite a bit (in comparison to vanilla nvim). It would be cool if packer.nvim could load the plugins asynchronysly or something like that.

**Comment by wbthomason** [comment]
"Faster" is hard to say with certainty. Fundamentally, `packer`, `plug`, and most every other plugin manager are doing the same thing - deferring sourcing plugins until certain events (i.e. lazy-loading). I like to think that `packer` makes it easier to do some more advanced forms of lazy-loading than `plug`, which could lead to improved load times, but this may vary by your use case. It is also worth noting that **more lazy-loading is not always better** - well-written plugins (which i.e. don't contain heavy logic in `plugin/` or `after/plugin/` that gets run on sourcing) often do not benefit significantly from being lazy-loaded, in which case you pay the (small) cost of the lazy-load logic for no benefit.

`packer`'s main novelty is the use of a pre-compiled file containing the logic for lazy-loading plugins, so that this doesn't need to be generated at startup. This saves a small amount of time. That said, `plug` (and `dein`) have the benefit of a longer existence and therefore much more development time/contributions from many more people.

Asynchronous loading would indeed be really cool, and something I'd love to add to `packer`. The challenge is that (1) loading concurrently is challenging because many plugins modify global state, and (2) loading asynchronously (but on a single thread) raises the question of how to ensure that plugins are sourced before they're needed.
